src/rules/ResourceQuoteStyle.js

```diff
@@ -216,33 +216,39 @@ class ResourceQuoteStyle extends ResourceRule {
             li = new LocaleInfo(locale);
             LICache[locale] = li;
         }
-    
+        /** special case for Japanese */
+        if (locale === "ja-JP") {
+            // accept the main quote style only and also square brackets
+            li.info.delimiter.alternateQuotationStart = "「\\[";
+            li.info.delimiter.alternateQuotationEnd = "」\\]";
```

Comment:
I think it would be clearer to apply overrides like:
```js
        const targetQuoteStart = li.getDelimiterQuotationStart();
        let targetQuoteStartAlt = /** @type {string} */ (li.info.delimiter.alternateQuotationStart);

        const targetQuoteEnd = li.getDelimiterQuotationEnd();
        let targetQuoteEndAlt = /** @type {string} */ (li.info.delimiter.alternateQuotationEnd);

        // special case for Japanese
        if (locale === "ja-JP") {
            // accept the main quote style only and also square brackets
            targetQuoteStartAlt = "「\\[";
            targetQuoteEndAlt = "」\\]";
        }
```

rather than modifying internals of an object of another library - only to pull it as-is a couple lines below